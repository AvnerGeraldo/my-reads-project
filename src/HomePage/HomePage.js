import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

//Components
import ButtonChooseShelf from '../ButtonChooseShelf/ButtonChooseShelf'
import Shelf from '../Shelf/Shelf'

class HomePage extends Component {

    state ={
        shelfActive: 'All'  
    }

    changeShelfHandler = (shelf) => {
        this.setState({ shelfActive: shelf })
    }

    render() {
        const listBookShelf = ['Currently Reading', 'Want to Read', 'Read', 'All']
        const { bookData, changeBookShelfHandler } = this.props

        return (
            <div>
                <div className="row align-items-center">
                    <div className="col list-books-title">
                        <h1>MyReads</h1>
                    </div>
                </div>
                <div className="row choose-bookshelf">
                    <div className="col-10 offset-1">
                        <div className="row">
                            {listBookShelf.map(shelf => (
                                <div key={shelf} className="col-3">
                                    <ButtonChooseShelf 
                                        bookshelf={shelf}
                                        changeShelf={this.changeShelfHandler}
                                        active={shelf === this.state.shelfActive}/>
                                </div>
                            ))}
                        </div>          
                    </div>
                    <div className="open-search">
                        <Link to="/search">Add a Book</Link>
                    </div>
                </div>
                <Shelf 
                    data={bookData} 
                    chooseShelf={this.state.shelfActive}
                    changeBookShelf={changeBookShelfHandler} />
            </div>
        )
    }
}

const { func, array } = PropTypes

HomePage.propTypes = {
    bookData: array.isRequired,
    changeBookShelfHandler: func.isRequired
}

export default HomePage