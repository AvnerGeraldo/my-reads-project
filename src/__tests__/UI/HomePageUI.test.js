import React from 'react'
import { shallow, mount } from 'enzyme'
import ReactRouterEnzymeContext from 'react-router-enzyme-context'

//Component
import HomePage from '../../HomePage/HomePage'
import { list } from '../../../node_modules/postcss';

const props = {
    bookData: [],
    changeBookShelfHandler: jest.fn()
}
const shelfCategoryList = ['Currently Reading', 'Want to Read', 'Read', 'All']
const options = new ReactRouterEnzymeContext();

describe('[Component] UI HomePage', () => {
    it('render correctly', () => {
        const wrapper = mount(<HomePage {...props}/>, options.get())
        expect(wrapper).toMatchSnapshot()
    })

    it('render the button choose shelf for every category of a shelf', () => {
        const wrapper = mount(<HomePage {...props}/>, options.get())
        const listRenderShelfs = []

        wrapper.find('.choose-bookshelf')
        .find('.row')
        .find('div.col-3').map(elemDiv => {
            listRenderShelfs.push(elemDiv.find('a').find('.title-shelf').text())
        })
        
        expect(shelfCategoryList).toEqual(expect.arrayContaining(listRenderShelfs))
    })

    it('verified if page link \'Add Book\' exists on HomePage', () => {
        const wrapper = mount(<HomePage {...props}/>, options.get())
        expect(wrapper.find('.open-search').find('a')).toHaveLength(1)
    })

    it('render shelf correctly', () => {
        const wrapper = mount(<HomePage {...props} />, options.get())
        expect(wrapper.find('.shelf-content')).toHaveLength(1)
    })
})