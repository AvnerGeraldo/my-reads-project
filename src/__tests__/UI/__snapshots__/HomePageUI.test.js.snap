// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] UI HomePage render correctly 1`] = `
<HomePage
  bookData={Array []}
  changeBookShelfHandler={[Function]}
>
  <div>
    <div
      className="row align-items-center"
    >
      <div
        className="col list-books-title"
      >
        <h1>
          MyReads
        </h1>
      </div>
    </div>
    <div
      className="row choose-bookshelf"
    >
      <div
        className="col-10 offset-1"
      >
        <div
          className="row"
        >
          <div
            className="col-3"
            key="Currently Reading"
          >
            <ButtonChooseShelf
              active={false}
              bookshelf="Currently Reading"
              changeShelf={[Function]}
            >
              <Link
                className={null}
                onClick={[Function]}
                replace={false}
                to="/"
              >
                <a
                  className={null}
                  href="/"
                  onClick={[Function]}
                >
                  <FontAwesomeIcon
                    border={false}
                    className="mx-auto d-block"
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          576,
                          512,
                          Array [],
                          "f518",
                          "M542.22 32.05c-54.8 3.11-163.72 14.43-230.96 55.59-4.64 2.84-7.27 7.89-7.27 13.17v363.87c0 11.55 12.63 18.85 23.28 13.49 69.18-34.82 169.23-44.32 218.7-46.92 16.89-.89 30.02-14.43 30.02-30.66V62.75c.01-17.71-15.35-31.74-33.77-30.7zM264.73 87.64C197.5 46.48 88.58 35.17 33.78 32.05 15.36 31.01 0 45.04 0 62.75V400.6c0 16.24 13.13 29.78 30.02 30.66 49.49 2.6 149.59 12.11 218.77 46.95 10.62 5.35 23.21-1.94 23.21-13.46V100.63c0-5.29-2.62-10.14-7.27-12.99z",
                        ],
                        "iconName": "book-open",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    transform={null}
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-book-open fa-w-18 mx-auto d-block"
                      data-icon="book-open"
                      data-prefix="fas"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M542.22 32.05c-54.8 3.11-163.72 14.43-230.96 55.59-4.64 2.84-7.27 7.89-7.27 13.17v363.87c0 11.55 12.63 18.85 23.28 13.49 69.18-34.82 169.23-44.32 218.7-46.92 16.89-.89 30.02-14.43 30.02-30.66V62.75c.01-17.71-15.35-31.74-33.77-30.7zM264.73 87.64C197.5 46.48 88.58 35.17 33.78 32.05 15.36 31.01 0 45.04 0 62.75V400.6c0 16.24 13.13 29.78 30.02 30.66 49.49 2.6 149.59 12.11 218.77 46.95 10.62 5.35 23.21-1.94 23.21-13.46V100.63c0-5.29-2.62-10.14-7.27-12.99z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </FontAwesomeIcon>
                  <div
                    className="title-shelf"
                  >
                    Currently Reading
                  </div>
                </a>
              </Link>
            </ButtonChooseShelf>
          </div>
          <div
            className="col-3"
            key="Want to Read"
          >
            <ButtonChooseShelf
              active={false}
              bookshelf="Want to Read"
              changeShelf={[Function]}
            >
              <Link
                className={null}
                onClick={[Function]}
                replace={false}
                to="/"
              >
                <a
                  className={null}
                  href="/"
                  onClick={[Function]}
                >
                  <FontAwesomeIcon
                    border={false}
                    className="mx-auto d-block"
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          448,
                          512,
                          Array [],
                          "f02d",
                          "M448 360V24c0-13.3-10.7-24-24-24H96C43 0 0 43 0 96v320c0 53 43 96 96 96h328c13.3 0 24-10.7 24-24v-16c0-7.5-3.5-14.3-8.9-18.7-4.2-15.4-4.2-59.3 0-74.7 5.4-4.3 8.9-11.1 8.9-18.6zM128 134c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm0 64c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm253.4 250H96c-17.7 0-32-14.3-32-32 0-17.6 14.4-32 32-32h285.4c-1.9 17.1-1.9 46.9 0 64z",
                        ],
                        "iconName": "book",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    transform={null}
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-book fa-w-14 mx-auto d-block"
                      data-icon="book"
                      data-prefix="fas"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 448 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M448 360V24c0-13.3-10.7-24-24-24H96C43 0 0 43 0 96v320c0 53 43 96 96 96h328c13.3 0 24-10.7 24-24v-16c0-7.5-3.5-14.3-8.9-18.7-4.2-15.4-4.2-59.3 0-74.7 5.4-4.3 8.9-11.1 8.9-18.6zM128 134c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm0 64c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm253.4 250H96c-17.7 0-32-14.3-32-32 0-17.6 14.4-32 32-32h285.4c-1.9 17.1-1.9 46.9 0 64z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </FontAwesomeIcon>
                  <div
                    className="title-shelf"
                  >
                    Want to Read
                  </div>
                </a>
              </Link>
            </ButtonChooseShelf>
          </div>
          <div
            className="col-3"
            key="Read"
          >
            <ButtonChooseShelf
              active={false}
              bookshelf="Read"
              changeShelf={[Function]}
            >
              <Link
                className={null}
                onClick={[Function]}
                replace={false}
                to="/"
              >
                <a
                  className={null}
                  href="/"
                  onClick={[Function]}
                >
                  <FontAwesomeIcon
                    border={false}
                    className="mx-auto d-block"
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          576,
                          512,
                          Array [],
                          "f530",
                          "M574.1 280.37L528.75 98.66c-5.91-23.7-21.59-44.05-43-55.81-21.44-11.73-46.97-14.11-70.19-6.33l-15.25 5.08c-8.39 2.79-12.92 11.86-10.12 20.24l5.06 15.18c2.79 8.38 11.85 12.91 20.23 10.12l13.18-4.39c10.87-3.62 23-3.57 33.16 1.73 10.29 5.37 17.57 14.56 20.37 25.82l38.46 153.82c-22.19-6.81-49.79-12.46-81.2-12.46-34.77 0-73.98 7.02-114.85 26.74h-73.18c-40.87-19.74-80.08-26.75-114.86-26.75-31.42 0-59.02 5.65-81.21 12.46l38.46-153.83c2.79-11.25 10.09-20.45 20.38-25.81 10.16-5.3 22.28-5.35 33.15-1.73l13.17 4.39c8.38 2.79 17.44-1.74 20.23-10.12l5.06-15.18c2.8-8.38-1.73-17.45-10.12-20.24l-15.25-5.08c-23.22-7.78-48.75-5.41-70.19 6.33-21.41 11.77-37.09 32.11-43 55.8L1.9 280.37A64.218 64.218 0 0 0 0 295.86v70.25C0 429.01 51.58 480 115.2 480h37.12c60.28 0 110.37-45.94 114.88-105.37l2.93-38.63h35.75l2.93 38.63C313.31 434.06 363.4 480 423.68 480h37.12c63.62 0 115.2-50.99 115.2-113.88v-70.25c0-5.23-.64-10.43-1.9-15.5zm-370.72 89.42c-1.97 25.91-24.4 46.21-51.06 46.21H115.2C86.97 416 64 393.62 64 366.11v-37.54c18.12-6.49 43.42-12.92 72.58-12.92 23.86 0 47.26 4.33 69.93 12.92l-3.13 41.22zM512 366.12c0 27.51-22.97 49.88-51.2 49.88h-37.12c-26.67 0-49.1-20.3-51.06-46.21l-3.13-41.22c22.67-8.59 46.08-12.92 69.95-12.92 29.12 0 54.43 6.44 72.55 12.93v37.54z",
                        ],
                        "iconName": "glasses",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    transform={null}
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-glasses fa-w-18 mx-auto d-block"
                      data-icon="glasses"
                      data-prefix="fas"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M574.1 280.37L528.75 98.66c-5.91-23.7-21.59-44.05-43-55.81-21.44-11.73-46.97-14.11-70.19-6.33l-15.25 5.08c-8.39 2.79-12.92 11.86-10.12 20.24l5.06 15.18c2.79 8.38 11.85 12.91 20.23 10.12l13.18-4.39c10.87-3.62 23-3.57 33.16 1.73 10.29 5.37 17.57 14.56 20.37 25.82l38.46 153.82c-22.19-6.81-49.79-12.46-81.2-12.46-34.77 0-73.98 7.02-114.85 26.74h-73.18c-40.87-19.74-80.08-26.75-114.86-26.75-31.42 0-59.02 5.65-81.21 12.46l38.46-153.83c2.79-11.25 10.09-20.45 20.38-25.81 10.16-5.3 22.28-5.35 33.15-1.73l13.17 4.39c8.38 2.79 17.44-1.74 20.23-10.12l5.06-15.18c2.8-8.38-1.73-17.45-10.12-20.24l-15.25-5.08c-23.22-7.78-48.75-5.41-70.19 6.33-21.41 11.77-37.09 32.11-43 55.8L1.9 280.37A64.218 64.218 0 0 0 0 295.86v70.25C0 429.01 51.58 480 115.2 480h37.12c60.28 0 110.37-45.94 114.88-105.37l2.93-38.63h35.75l2.93 38.63C313.31 434.06 363.4 480 423.68 480h37.12c63.62 0 115.2-50.99 115.2-113.88v-70.25c0-5.23-.64-10.43-1.9-15.5zm-370.72 89.42c-1.97 25.91-24.4 46.21-51.06 46.21H115.2C86.97 416 64 393.62 64 366.11v-37.54c18.12-6.49 43.42-12.92 72.58-12.92 23.86 0 47.26 4.33 69.93 12.92l-3.13 41.22zM512 366.12c0 27.51-22.97 49.88-51.2 49.88h-37.12c-26.67 0-49.1-20.3-51.06-46.21l-3.13-41.22c22.67-8.59 46.08-12.92 69.95-12.92 29.12 0 54.43 6.44 72.55 12.93v37.54z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </FontAwesomeIcon>
                  <div
                    className="title-shelf"
                  >
                    Read
                  </div>
                </a>
              </Link>
            </ButtonChooseShelf>
          </div>
          <div
            className="col-3"
            key="All"
          >
            <ButtonChooseShelf
              active={true}
              bookshelf="All"
              changeShelf={[Function]}
            >
              <Link
                className="active"
                onClick={[Function]}
                replace={false}
                to="/"
              >
                <a
                  className="active"
                  href="/"
                  onClick={[Function]}
                >
                  <FontAwesomeIcon
                    border={false}
                    className="mx-auto d-block"
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          448,
                          512,
                          Array [],
                          "f558",
                          "M318.38 208h-39.09c-1.49 27.03-6.54 51.35-14.21 70.41 27.71-13.24 48.02-39.19 53.3-70.41zm0-32c-5.29-31.22-25.59-57.17-53.3-70.41 7.68 19.06 12.72 43.38 14.21 70.41h39.09zM224 97.31c-7.69 7.45-20.77 34.42-23.43 78.69h46.87c-2.67-44.26-15.75-71.24-23.44-78.69zm-41.08 8.28c-27.71 13.24-48.02 39.19-53.3 70.41h39.09c1.49-27.03 6.53-51.35 14.21-70.41zm0 172.82c-7.68-19.06-12.72-43.38-14.21-70.41h-39.09c5.28 31.22 25.59 57.17 53.3 70.41zM247.43 208h-46.87c2.66 44.26 15.74 71.24 23.43 78.69 7.7-7.45 20.78-34.43 23.44-78.69zM448 358.4V25.6c0-16-9.6-25.6-25.6-25.6H96C41.6 0 0 41.6 0 96v320c0 54.4 41.6 96 96 96h326.4c12.8 0 25.6-9.6 25.6-25.6v-16c0-6.4-3.2-12.8-9.6-19.2-3.2-16-3.2-60.8 0-73.6 6.4-3.2 9.6-9.6 9.6-19.2zM224 64c70.69 0 128 57.31 128 128s-57.31 128-128 128S96 262.69 96 192 153.31 64 224 64zm160 384H96c-19.2 0-32-12.8-32-32s16-32 32-32h288v64z",
                        ],
                        "iconName": "atlas",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    transform={null}
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-atlas fa-w-14 mx-auto d-block"
                      data-icon="atlas"
                      data-prefix="fas"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 448 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M318.38 208h-39.09c-1.49 27.03-6.54 51.35-14.21 70.41 27.71-13.24 48.02-39.19 53.3-70.41zm0-32c-5.29-31.22-25.59-57.17-53.3-70.41 7.68 19.06 12.72 43.38 14.21 70.41h39.09zM224 97.31c-7.69 7.45-20.77 34.42-23.43 78.69h46.87c-2.67-44.26-15.75-71.24-23.44-78.69zm-41.08 8.28c-27.71 13.24-48.02 39.19-53.3 70.41h39.09c1.49-27.03 6.53-51.35 14.21-70.41zm0 172.82c-7.68-19.06-12.72-43.38-14.21-70.41h-39.09c5.28 31.22 25.59 57.17 53.3 70.41zM247.43 208h-46.87c2.66 44.26 15.74 71.24 23.43 78.69 7.7-7.45 20.78-34.43 23.44-78.69zM448 358.4V25.6c0-16-9.6-25.6-25.6-25.6H96C41.6 0 0 41.6 0 96v320c0 54.4 41.6 96 96 96h326.4c12.8 0 25.6-9.6 25.6-25.6v-16c0-6.4-3.2-12.8-9.6-19.2-3.2-16-3.2-60.8 0-73.6 6.4-3.2 9.6-9.6 9.6-19.2zM224 64c70.69 0 128 57.31 128 128s-57.31 128-128 128S96 262.69 96 192 153.31 64 224 64zm160 384H96c-19.2 0-32-12.8-32-32s16-32 32-32h288v64z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </FontAwesomeIcon>
                  <div
                    className="title-shelf"
                  >
                    All
                  </div>
                </a>
              </Link>
            </ButtonChooseShelf>
          </div>
        </div>
      </div>
      <div
        className="open-search"
      >
        <Link
          replace={false}
          to="/search"
        >
          <a
            href="/search"
            onClick={[Function]}
          >
            Add a Book
          </a>
        </Link>
      </div>
    </div>
    <Shelf
      changeBookShelf={[Function]}
      chooseShelf="All"
      data={Array []}
    >
      <div
        className="row shelf-content"
        style={
          Object {
            "margin": "0 auto",
            "maxWidth": "800px",
            "minWidth": "675px",
            "padding": "0px 10px",
          }
        }
      >
        <div
          className="no-results"
        >
          Não há dados para serem exibidos!
        </div>
      </div>
    </Shelf>
  </div>
</HomePage>
`;
